### DISCUSSION TITLE
How can I prevent all fetches, if the internet connection lost?

### DISCUSSION THREAD

**Discussion by bozdevs** [swup/swup#903] [General]
The internet connection when lost, when you click a link it route page with browser method (and it response "`no internet connection...`"). But I want stay last "`200`" response and pop a message. How can I prevent all fetches, if the internet connection lost?

I can't prevent event because the `swup:fetch:error` hook doesn't fire when no internet.

**Comment by daun** [comment]
There's currently nothing built in to prevent or cancel a fetch from a hook. There's a recent [pull request for an abort/undo feature](https://github.com/swup/swup/pull/850) which would cover your use case, but we decided against implementing it for the time being as it brings little benefit for the amount of complexity added. That said, I'd consider this a valid feature request to make swup aware of the browser's network status and ignore any visits while offline.

**Reply by bozdevs** [reply]
Thanks +1
